### PR TITLE
Add organization search

### PR DESCRIPTION
**Pull request by jakub-lotko**
- add pagination control
- add tests

### CODE REVIEW

**Review by lukasz-ciura-pwc — CHANGES_REQUESTED** [review]

File: src/HE.Investments.Organisation/Config/ServiceCollectionExtensions.cs
Comment (lukasz-ciura-pwc) [inline src/HE.Investments.Organisation/Config/ServiceCollectionExtensions.cs:1]: ❤

File: src/HE.Investments.Organisation/Services/IOrganizationCrmSearchService.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using HE.Common.IntegrationModel.PortalIntegrationModel;
+using HE.Investments.Organisation.CrmRepository;
+using Microsoft.PowerPlatform.Dataverse.Client;
+
+namespace HE.Investments.Organisation.Services;
+internal interface IOrganizationCrmSearchService
```
Comment (lukasz-ciura-pwc) [inline src/HE.Investments.Organisation/Services/IOrganizationCrmSearchService.cs:6]: Put interface to separate file.

File: src/HE.Investments.Organisation/Services/OrganisationSearchService.cs [hunk 2]
```diff
@@ -1,53 +1,72 @@
...
+
+        var organizationsFromCrm = GetMatchingOrganizationsFromCrm(companyHousesOrganizations);
+
+        var mergedResult = MergeResultes(companyHousesOrganizations, organizationsFromCrm);
+
+        return new OrganisationSearchResult(mergedResult, companyHousesResult.TotalItems, null!);
     }
 
-    public List<OrganizationDetailsDto> SearchOrganizationInCrm(List<string> organisationNumbers, IOrganizationServiceAsync2 service)
+    private async Task<OrganisationSearchResult> GetOrganizationFromCompenyHousesApi(string organisationName, PagingQueryParams pagingParams, CancellationToken cancellationToken)
```
Comment (lukasz-ciura-pwc) [inline src/HE.Investments.Organisation/Services/OrganisationSearchService.cs:37]: Typo GetOrganizationFrom**Compeny**HousesApi

File: src/HE.Investments.Organisation/Services/OrganisationSearchService.cs [hunk 3]
```diff
@@ -1,53 +1,72 @@
...
-        var retrievedEntities = _organizationRepository.SearchForOrganizations(service, organisationNumbers);
-        if (retrievedEntities != null && retrievedEntities.Entities.Count > 0)
+        try
+        {
+            var companyHouseApiResult = await _companiesHouseApi.Search(organisationName, pagingParams, cancellationToken);
+
+            return new OrganisationSearchResult(companyHouseApiResult.Items.Select(x => new OrganisationSearchItem(x.CompanyNumber, x.CompanyName, x.OfficeAddress.Locality!, x.OfficeAddress.AddressLine1!, x.OfficeAddress.PostalCode!)).ToList(), companyHouseApiResult.Hits, null!);
+        }
+        catch (HttpRequestException ex)
         {
```
Comment (lukasz-ciura-pwc) [inline src/HE.Investments.Organisation/Services/OrganisationSearchService.cs:46]: Exception should be logged.

**Review by lukasz-ciura-pwc — APPROVED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/Organization/QueryHandlers/SearchOrganizationsHandler.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+using MediatR;
+using Org.HE.Investments.Organisation.CompaniesHouse.Contract;
+using Org.HE.Investments.Organisation.Services;
+
+namespace HE.InvestmentLoans.BusinessLogic.Organization.QueryHandlers;
+internal class SearchOrganizationsHandler : IRequestHandler<SearchOrganizationsQuery, SearchOrganisationsQueryResponse>
+{
+    private readonly IOrganisationSearchService _searchService;
+
+    public SearchOrganizationsHandler(IOrganisationSearchService searchService)
```
Comment (dawid-pawulski) [inline src/HE.InvestmentLoans.BusinessLogic/Organization/QueryHandlers/SearchOrganizationsHandler.cs:15]: We can change that to SearchOrganizationsQueryHandler

File: src/HE.InvestmentLoans.WWW/Controllers/OrganizationController.cs [hunk 5]
```diff
@@ -21,11 +24,24 @@ public IActionResult SearchOrganization()
         return View();
     }
 
+    [HttpPost("search")]
+    public IActionResult SearchOrganizationPost(OrganizationViewModel organization)
```
Comment (dawid-pawulski) [inline src/HE.InvestmentLoans.WWW/Controllers/OrganizationController.cs:28]: To be consistent with other controllers we don't need to add "Post" at the end of this method name

File: src/He.AspNetCore.Mvc.Gds.Components/TagHelpers/Pagination/GdsPaginationTagHelper.cs [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+
+            if (output == null)
+            {
+                return;
+            }
+
+            output.TagName = "nav";
```
Comment (dawid-pawulski) [inline src/He.AspNetCore.Mvc.Gds.Components/TagHelpers/Pagination/GdsPaginationTagHelper.cs:31]: We can add strings from this file to constants

**Review by dawid-pawulski — APPROVED** [review]

**Pull request merged**